client/hotrod-client/src/main/java/org/infinispan/client/hotrod/telemetry/impl/TelemetryServiceFactory.java

```diff
@@ -0,0 +1,49 @@
+package org.infinispan.client.hotrod.telemetry.impl;
+
+import org.infinispan.client.hotrod.logging.Log;
+import org.infinispan.client.hotrod.logging.LogFactory;
+
+public class TelemetryServiceFactory {
+
+   private static final Log log = LogFactory.getLog(TelemetryService.class, Log.class);
+
+   public static final TelemetryServiceFactory INSTANCE = new TelemetryServiceFactory();
+
+   private final TelemetryService telemetryService;
+   private volatile boolean loggedDisabledTracing = false;
+
+   private TelemetryServiceFactory() {
+      telemetryService = telemetryService();
+   }
+
+   public TelemetryService telemetryService(boolean tracingPropagationEnabled) {
+      if (tracingPropagationEnabled) {
+         return telemetryService;
+      }
+
+      if (loggedDisabledTracing == false) {
+         loggedDisabledTracing = true;
+         log.openTelemetryPropagationDisabled();
```

Comment:
JBoss Logging has a  `@Once` annotation. See https://github.com/infinispan/infinispan/blob/main/server/hotrod/src/main/java/org/infinispan/server/hotrod/logging/Log.java#L56 for an example

Reply:
cool